compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/ProjectLoader.java

```diff
@@ -95,31 +136,41 @@ public static Project loadProject(Path path, ProjectEnvironmentBuilder projectEn
      * @return documentId of the document
      */
 
-    public static Optional<DocumentId> getDocumentId(Path documentFilePath, BuildProject project) {
+    public static Document getDocument(Path documentFilePath, Project project) {
+        if (project.kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
+            DocumentId documentId = project.currentPackage().getDefaultModule().documentIds().iterator().next();
+            return project.currentPackage().getDefaultModule().document(documentId);
+        }
+        if (!(project instanceof BuildProject)) {
+            throw new ProjectException("unsupported project type:" + project.kind());
+        }
+        BuildProject buildProject = (BuildProject) project;
         Path parent = Optional.of(documentFilePath.getParent()).get();
+
         for (ModuleId moduleId : project.currentPackage().moduleIds()) {
-            Optional<Path> modulePath = project.modulePath(moduleId);
-            if (modulePath.isPresent()) {
-                if (parent.equals(modulePath.get())
-                        || parent.equals(modulePath.get().resolve(ProjectConstants.TEST_DIR_NAME))) {
-                    Module module = project.currentPackage().module(moduleId);
-                    for (DocumentId documentId : module.documentIds()) {
-                        if (module.document(documentId).name().equals(
-                                Optional.of(documentFilePath.getFileName()).get().toString())) {
-                            return Optional.of(documentId);
-                        }
+            Optional<Path> modulePath = buildProject.modulePath(moduleId);
+            if (modulePath.isEmpty()) {
+                throw new ProjectException("document path does to belong to any module");
+            }
+            if (parent.equals(modulePath.get())
+                    || parent.equals(modulePath.get().resolve(ProjectConstants.TEST_DIR_NAME))) {
+                Module module = project.currentPackage().module(moduleId);
+                for (DocumentId documentId : module.documentIds()) {
+                    if (module.document(documentId).name().equals(
+                            Optional.of(documentFilePath.getFileName()).get().toString())) {
+                        return module.document(documentId);
                     }
+                }
 
-                    for (DocumentId documentId : module.testDocumentIds()) {
-                        if (module.document(documentId).name().equals(ProjectConstants.TEST_DIR_NAME + "/"
-                                + Optional.of(documentFilePath.getFileName()).get().toString())) {
-                            return Optional.of(documentId);
-                        }
+                for (DocumentId documentId : module.testDocumentIds()) {
+                    if (module.document(documentId).name().equals(ProjectConstants.TEST_DIR_NAME + "/"
```

Comment:
Can you verify whether use of '/' here is platform independent. If yes, shall we move that into a more convincing constant such as `TEST_FILE_SEPARATOR`